modules/carousel/jetpack-carousel.php

```diff
@@ -275,22 +301,56 @@ function enqueue_assets( $output ) {
 
 			$this->first_run = false;
 		}
-
-		return $output;
 	}
 
 	function set_in_gallery( $output ) {
 		$this->in_gallery = true;
 		return $output;
 	}
 
-	function add_data_to_images( $attr, $attachment = null ) {
+	/**
+	 * Adds data-* attributes required by carousel to img tags in post HTML
+	 * content. To be used by 'the_content' filter.
+	 *
+	 * @see add_data_to_images()
+	 * @see wp_make_content_images_responsive() in wp-includes/media.php
+	 *
+	 * @param string $content HTML content of the post
+	 * @return string Modified HTML content of the post
+	 */
+	function add_data_to_single_images( $content ) {
```

Comment:
Even though this function is similarly named as `add_data_to_images` they do very different things – this one parses the HTML and adds HTML attributes, while the other one works only on structured data.